### PR TITLE
Fix/test suite

### PR DESCRIPTION
**Pull request by lfwa**
This PR fixes the failing test suite:
- Removes `torchaudio` dependency used in automatic-speech-recognition in favor of `soundfile` by upgrading `datasets` to the latest version 2.12.0 due to failing tests (except for macOS running x86_64 on GitHub Action as the prepackaged `libsndfile` in `soundfile>=0.12.0` gives different results, and we therefore have to downgrade `soundfile` to version 0.11.0 that uses the system version of `libsndfile`).
- Fixes failing `test_evaluator.py/test_evaluate_single[automatic-speech-recognition]` by fixing above and updating expected output.
- Temporarily skips leaderboard tests until it is back online.
- Cleans up CI workflow.

### CODE REVIEW

File: src/alexandra_ai_eval/evaluator.py [hunk 1]
```diff
@@ -325,6 +328,12 @@ def _send_results_to_leaderboard(self) -> List[bool]:
                 A list of booleans indicating whether the results were
                 successfully sent to the leaderboard.
         """
+        assert (
+            self.leaderboard_client is not None
+        ), "Leaderboard client is not initialized. Ensure that self.send_results_to_leaderboard is set to True."
```
Comment (saattrupdan) [inline src/alexandra_ai_eval/evaluator.py:333]: Maybe mention that the class needs to be re-initialised. I.e., it's not enough to just set `self.send_results_to_leaderboard` to True.

File: src/alexandra_ai_eval/leaderboard_utils.py [hunk 2]
```diff
@@ -167,3 +167,17 @@ def post_model_to_task(
         response.close()
 
         return response_json
+
+    def check_connection(self, timeout: int = 5):
+        """Checks whether we can establish a connection to the leaderboard.
+
+        Args:
+            timeout (int): The timeout in seconds. Defaults to 5.
```
Comment (saattrupdan) [inline src/alexandra_ai_eval/leaderboard_utils.py:175]:
```suggestion
            timeout (int, optional): 
                The timeout in seconds. Defaults to 5.
```

**Review by saattrupdan — CHANGES_REQUESTED** [review]
Looks good! 😊  Only have a few minor changes.

**Comment by lfwa** [comment]
We should probably wait with merging until the CI passes. Maybe we should consider blocking all PRs with non-passed checks to ensure that our test suite does not grow stale or we accidentally introduce bugs.

**Review by saattrupdan — APPROVED** [review]
LGTM! Also added the status check branch protection rule now, good idea 👍

**Comment by AJDERS** [comment]
I agree with you idea @lfwa that we should move the ASR-part to be optional. I think for now we should re-include `torchaudio`, and merge this PR asap, and then deal with making the ASR-part optional in another PR. Experience shows that any features caught on a branch with the current ASR-bug will be there for quite some time, so we should avoid that.

**Comment by lfwa** [comment]
@AJDERS Good thinking. I am inclined to agree, but reintroducing `torchaudio` likely won't fix things after upgrading the `datasets` dependency since they entirely removed `torchaudio` in favor of `soundfile` in version 2.11. And if we downgrade `datasets` again then the test suite will fail nonetheless.

I will devote a bit of time to see if I can fix this, probably add a few commits to this PR just to test the CI. If it takes too long then it may be better to just merge this PR as is and then start work on making ASR optional as you said.

File: .github/workflows/ci.yaml [hunk 3]
```diff
@@ -49,6 +35,17 @@ jobs:
...
+      - name: Downgrade soundfile version on macOS x86_64
+        # soundfile>=0.12.0 prepackages libsndfile, but the prepackaged version gives
+        # different results than the system version for macOS running x86_64
+        # (which is what GitHub Actions uses). As such we have to downgrade to
+        # soundfile==0.11.0 and use the system libsndfile.
+        run: |
+          if [ "$RUNNER_OS" == "macOS" ] && [ "$(uname -m)" == "x86_64" ]; then
+            poetry add soundfile==0.11.0
+          fi
+        shell: bash
```
Comment (saattrupdan) [inline .github/workflows/ci.yaml:47]: Does this mean that the MacOs server had `libsndfile` pre-installed all along?

**Review by saattrupdan — COMMENTED** [review]
LGTM! I just have a single question, which is more for the sake of my own understanding.

File: .github/workflows/ci.yaml (hunk 3 above)
Comment (lfwa) [reply .github/workflows/ci.yaml:47]: Yea, looks like it. It is installed in our macOS environment at least. We cannot be sure if it breaks in the future, but hopefully `soundfile` will include another `libsndfile` binary for Darwin x86_64 in the future that gives the same results as on Silicon and Ubuntu and we won't have to downgrade the version in the workflow.

**Comment by lfwa** [comment]
@saattrupdan Do you know why merging is blocked?
It says "The base branch does not allow updates," but I am not exactly sure what this means or how I fix it.

**Comment by saattrupdan** [comment]
> @saattrupdan Do you know why merging is blocked? It says "The base branch does not allow updates," but I am not exactly sure what this means or how I fix it.

My bad, it was too strict a branch protection policy. Fixed now - merging.

**Pull request merged**